server.go

```diff
@@ -492,18 +494,50 @@ func deadline(timeout time.Duration) time.Time {
 }
 
 func readRequest(ctx context.Context, conn *serverConn, done chan<- error) (*Request, error) {
+	var cmd Command
+
 	args := newByteArgsReader(&conn.p, done)
 
-	req := &Request{
-		Addr:    conn.RemoteAddr().String(),
-		Args:    args,
-		Context: ctx,
+	cmd = Command{
```

Comment:
How about doing `cmd := Command{` here and get rid of the variable declaration above?